hs_core/search_indexes.py

```diff
@@ -51,49 +51,71 @@ class BaseResourceIndex(indexes.SearchIndex, indexes.Indexable):
     editors_names = indexes.MultiValueField(faceted=True)
     editors_count = indexes.IntegerField(faceted=True)
 
-
     def get_model(self):
         return BaseResource
 
     def index_queryset(self, using=None):
-        return self.get_model().objects.filter(Q(raccess__discoverable=True) | Q(raccess__public=True))
+        return self.get_model().objects.filter(Q(raccess__discoverable=True) |
+                                               Q(raccess__public=True))
 
     def prepare_title(self, obj):
-        if hasattr(obj, 'metadata'):
+        if hasattr(obj, 'metadata') and obj.metadata.title.value is not None:
             return obj.metadata.title.value
         else:
             return 'none'
 
+    def prepare_abstract(self, obj):
+        if hasattr(obj, 'metadata') and obj.metadata.description.abstract is not None:
```

Comment:
Actually, it can't.... If it's none, the code would raise an exception on accessing BaseResource.description... execution would never get here. See implementation of the @property description for details.